src/NHibernate/Cache/SyncCacheLock.cs

```diff
@@ -10,28 +10,23 @@ class SyncCacheLock : ICacheLock
 
 		class MonitorLock : IDisposable
```

Comment:
That is no more a Monitor based lock with your change. Better change its name. `InternalLock`, maybe?

Reply:
Done